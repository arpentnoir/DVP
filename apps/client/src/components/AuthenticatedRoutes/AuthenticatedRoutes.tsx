import { Authenticator } from '@aws-amplify/ui-react';
import { FunctionComponent } from 'react';
import { getEnvConfig } from '../../config';
import { Outlet } from 'react-router-dom';

const envConfig = getEnvConfig();

export const AuthenticatedRoutes: FunctionComponent = () => {
  return (
    <div style={{ height: '100%' }}>
      <Authenticator variation="modal" hideSignUp={envConfig.DISABLE_SIGNUP}>
        <Outlet />
      </Authenticator>
    </div>
  );
};
