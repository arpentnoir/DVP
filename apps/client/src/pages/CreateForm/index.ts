export * from './AANZFTACOOSchema';
export * from './CreateFormPage';
