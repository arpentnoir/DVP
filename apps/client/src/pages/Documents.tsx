import { DataTable, MoreInfo, Text } from '@dvp/vc-ui';
import { Box } from '@mui/material';
import { GridColDef } from '@mui/x-data-grid';
import { useNavigate } from 'react-router-dom';
import { BaseLayout } from '../layouts';

// TODO: Replace with real data
const columns: GridColDef[] = [
  {
    field: 'id',
    headerName: 'ID',
    minWidth: 200,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'ABN',
    headerName: 'ABN',
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'businessName',
    headerName: 'Business name',
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'document',
    headerName: 'Document',
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'dateIssued',
    headerName: 'Date Issued',
    sortable: false,
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'moreInfo',
    headerName: '',
    flex: 1,
    renderCell: (params) => (
      <MoreInfo
        params={params}
        items={{ View: () => null, Download: () => null, Revoke: () => null }}
      />
    ),
    disableColumnMenu: true,
    sortable: false,
  },
];

const rows = [
  {
    id: '5940107005302237590',
    document: 'AANZFTA',
    businessName: 'Organisation',
    ABN: '1234 5678 90',
    dateIssued: '01/01/2012',
  },
  {
    id: '5940107005302237591',
    document: 'CITES',
    businessName: 'Organisation',
    ABN: '1234 5678 91',
    dateIssued: '01/01/2012',
  },
  {
    id: '5940107005302237592',
    document: 'AANZFTA',
    businessName: 'Organisation',
    ABN: '1234 5678 92',
    dateIssued: '01/01/2018',
  },
  {
    id: '5940107005302237593',
    document: 'CITES',
    businessName: 'Organisation',
    ABN: '1234 5678 93',
    dateIssued: '01/01/2018',
  },
  {
    id: '5940107005302237594',
    document: 'CITES',
    businessName: 'Organisation',
    ABN: '1234 5678 94',
    dateIssued: '01/01/2032',
  },
  {
    id: '5940107005302237595',
    document: 'AANZFTA',
    businessName: 'Organisation',
    ABN: '1234 5678 95',
    dateIssued: '01/01/2019',
  },
  {
    id: '5940107005302237596',
    document: 'CITES',
    businessName: 'Organisation',
    ABN: '1234 5678 96',
    dateIssued: '01/01/2019',
  },
  {
    id: '5940107005302237597',
    document: 'CITES',
    businessName: 'Organisation',
    ABN: '1234 5678 97',
    dateIssued: '01/01/2020',
  },
  {
    id: '5940107005302237598',
    document: 'AANZFTA',
    businessName: 'Organisation',
    ABN: '1234 5678 98',
    dateIssued: '01/01/2022',
  },
];

export const Documents = () => {
  const navigate = useNavigate();

  const handleIssue = () => {
    navigate('/issue');
  };

  return (
    <BaseLayout title="Documents">
      <Box paddingBottom="40px">
        <Text variant="h1" fontWeight="bold" paddingBottom="24px">
          Documents
        </Text>
        <Text>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu
          turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec
          fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus
          elit sed risus. Maecenas eget condimentum velit, sit amet feugiat
          lectus. Class aptent taciti sociosqu ad litora torquent per conubia
          nostra, per inceptos himenaeos. Praesent auctor purus luctus enim
          egestas, ac scelerisque ante pulvinar. Donec ut rhoncus ex.
          Suspendisse ac rhoncus nisl, eu tempor urna. Curabitur vel bibendum
          lorem. Morbi convallis convallis diam sit amet lacinia. Aliquam in
          elementum tellus.
        </Text>
        <Text variant="h3" fontWeight="bold" paddingTop="40px">
          Documents issued
        </Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</Text>
      </Box>

      <DataTable
        handleAction={handleIssue}
        toolBarActionLabel="Issue"
        rows={rows}
        columns={columns}
        pageSize={5}
        rowsPerPageOptions={[5]}
        checkboxSelection
      />
    </BaseLayout>
  );
};
