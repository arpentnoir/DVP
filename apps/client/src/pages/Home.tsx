import { Card, Text } from '@dvp/vc-ui';
import NavigateNextIcon from '@mui/icons-material/NavigateNext';
import { Box, ButtonBase, Grid, Stack, useTheme } from '@mui/material';
import { useNavigate } from 'react-router-dom';
import { ROUTES } from '../constants';
import { BaseLayout } from '../layouts';

// TODO: Replace with real data
const actions = [
  {
    name: 'Lorem',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
  },
  {
    name: 'Ipsum',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
  },
  {
    name: 'Dolor',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
  },
  {
    name: 'Amet',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
  },
];

export const Home = () => {
  const theme = useTheme();
  const navigate = useNavigate();

  return (
    <BaseLayout
      title="Digital Verification Platform"
      sx={{
        paddingX: 0,
        paddingY: 0,
        backgroundColor: theme.palette.common.backgroundBlue,
        display: 'flex',
        justifyContent: 'center',
        maxWidth: '100%',
      }}
    >
      <Grid container height={'100%'} maxWidth={'1140px'}>
        <Stack>
          <Stack
            direction={'row'}
            sx={{
              height: 'calc(50vw * 0.62)',
              minHeight: { xs: '370px', md: '410px' },
              maxHeight: '450px',
              width: '100%',
            }}
          >
            <Grid item xs={12} sm={6}>
              <Stack
                padding={{
                  xs: '60px 20px 0px 20px',
                  sm: '40px 30px',
                  md: '60px 0px 0px 30px',
                }}
              >
                <Box>
                  <Text variant="h1" fontWeight="bold" color="white">
                    DIGITAL VERIFICATION
                  </Text>
                  <Text variant="h1" fontWeight="bold" color="white">
                    PLATFORM
                  </Text>
                </Box>
                <Text color="white" paddingTop="12px">
                  Lorem ipsum dolar sit amet consectetur
                </Text>
                <Stack padding="8px 10px 0 0">
                  <ButtonBase onClick={() => navigate(ROUTES.VERIFY)}>
                    <Stack
                      width="100%"
                      direction="row"
                      justifyContent="space-between"
                      alignItems="center"
                      sx={{
                        borderBottom: '1px solid white',
                        padding: '16px 20px',
                      }}
                    >
                      <Text color="white" fontWeight="bold" variant="body1">
                        Verify a document
                      </Text>
                      <NavigateNextIcon
                        sx={{ color: theme.palette.common.featureGold }}
                      />
                    </Stack>
                  </ButtonBase>
                  <ButtonBase>
                    <Stack
                      width="100%"
                      direction="row"
                      justifyContent="space-between"
                      alignItems="center"
                      sx={{
                        borderBottom: '1px solid white',
                        padding: '16px 20px',
                      }}
                    >
                      <Text color="white" fontWeight="bold" variant="body1">
                        Help and support
                      </Text>
                      <NavigateNextIcon
                        sx={{ color: theme.palette.common.featureGold }}
                      />
                    </Stack>
                  </ButtonBase>
                </Stack>
              </Stack>
            </Grid>
            <Grid
              item
              sm={6}
              sx={{
                display: {
                  xs: 'none',
                  sm: 'block',
                  background: 'url("assets/home-background.jpg")',
                  backgroundPosition: 'bottom right',
                  backgroundSize: 'cover',
                  backgroundRepeat: 'no-repeat',
                  overflow: 'hidden',
                },
                position: 'absolute',
                width: '100%',
                right: 0,
                height: 'calc(50vw * 0.62)',
                minHeight: { xs: '370px', md: '410px' },
                maxHeight: '450px',
              }}
            >
              <Box
                sx={{
                  background: theme.palette.common.backgroundBlue,
                  position: 'relative',
                  top: '-100px',
                  left: '-120px',
                  width: '200px',
                  height: 'calc(100% + 300px)',
                  transform: 'rotate(-15deg)',
                }}
              />
            </Grid>
          </Stack>
          <Grid
            container
            spacing="30px"
            padding={{
              xs: '0px 40px 100px 40px',
              sm: '60px 40px',
            }}
          >
            {actions.map((action) => (
              <Grid item lg={3} md={4} sm={6} xs={12} key={action.name}>
                <Card
                  sx={{ flex: 1, padding: '16px' }}
                  name={action.name}
                  text={action.text}
                  handleAction={() => null}
                  actionLabel="Action"
                />
              </Grid>
            ))}
          </Grid>
        </Stack>
      </Grid>
    </BaseLayout>
  );
};
