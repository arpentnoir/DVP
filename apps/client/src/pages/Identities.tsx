import { Button, Card, DataTable, MoreInfo, Text } from '@dvp/vc-ui';
import AddIcon from '@mui/icons-material/Add';
import { Box, Stack } from '@mui/material';
import { GridColDef } from '@mui/x-data-grid';
import { BaseLayout } from '../layouts';

// TODO: Replace with real data
const columns: GridColDef[] = [
  {
    field: 'nickname',
    headerName: 'Nickname',
    minWidth: 200,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'id',
    headerName: 'Public key',
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'dateIssued',
    headerName: 'Date Issued',
    sortable: false,
    minWidth: 150,
    flex: 1,
    disableColumnMenu: true,
  },
  {
    field: 'moreInfo',
    headerName: '',
    flex: 1,
    renderCell: (params) => (
      <MoreInfo
        params={params}
        items={{ Remove: () => null, Revoke: () => null }}
      />
    ),
    disableColumnMenu: true,
    sortable: false,
  },
];

const rows = [
  {
    id: '5940107005302237590',
    nickname: 'ABC Internal',
    dateIssued: '01/01/2012',
  },
  {
    id: '5940107005302237591',
    nickname: 'ABC External',
    dateIssued: '01/01/2012',
  },
  {
    id: '5940107005302237592',
    nickname: 'XYZ Private',
    dateIssued: '01/01/2018',
  },
  {
    id: '5940107005302237593',
    nickname: 'XYZ Public',
    dateIssued: '01/01/2018',
  },
];

const identities = [
  {
    name: 'Identity 1',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
    handleAction: () => null,
  },
  {
    name: 'Identity 2',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
    handleAction: () => null,
  },
  {
    name: 'Identity 3',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
    handleAction: () => null,
  },
];

export const Identities = () => {
  const handleCreateNew = () => null;

  return (
    <BaseLayout title="Identities">
      <Box paddingBottom="40px">
        <Text variant="h4" fontWeight="bold" paddingBottom="24px">
          Identities
        </Text>
        <Text>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu
          turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec
          fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus
          elit sed risus. Maecenas eget condimentum velit, sit amet feugiat
          lectus. Class aptent taciti sociosqu ad litora torquent per conubia
          nostra, per inceptos himenaeos. Praesent auctor purus luctus enim
          egestas, ac scelerisque ante pulvinar. Donec ut rhoncus ex.
          Suspendisse ac rhoncus nisl, eu tempor urna. Curabitur vel bibendum
          lorem. Morbi convallis convallis diam sit amet lacinia. Aliquam in
          elementum tellus.
        </Text>
      </Box>

      <Box paddingBottom="40px">
        <Text variant="body1" fontWeight="bold">
          Identifiers (DIDs)
        </Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</Text>
        <Stack
          direction={{ xs: 'column', sm: 'row' }}
          spacing="8px"
          paddingY="40px"
        >
          {identities.map((identity) => (
            <Card
              key={identity.name}
              name={identity.name}
              text={identity.text}
              handleAction={identity.handleAction}
              actionLabel="Remove"
            />
          ))}
        </Stack>
        <Stack direction="row" spacing="16px">
          <Button label="Upload DID" variant="outlined" />
          <Button label="Create new" leftIcon={<AddIcon />} color="error" />
        </Stack>
      </Box>

      <Box paddingBottom="30px">
        <Text variant="body1" fontWeight="bold">
          Key pairs
        </Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</Text>
      </Box>

      <DataTable
        handleAction={handleCreateNew}
        toolBarActionLabel="Create new"
        rows={rows}
        columns={columns}
        pageSize={5}
        rowsPerPageOptions={[5]}
        checkboxSelection
      />
    </BaseLayout>
  );
};
