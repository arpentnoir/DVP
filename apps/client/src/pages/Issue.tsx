import { Form, FormOption, Text, VCOptions } from '@dvp/vc-ui';
import { Grid, Paper } from '@mui/material';
import { useNavigate } from 'react-router-dom';
import { BaseLayout } from '../layouts';
import {
  schema as partialCooSchema,
  schemaUI as partialCooSchemaUi,
} from './CreateForm/AANZFTACOOPartialSchema';
import {
  schema as fullCooSchema,
  schemaUI as fullCooSchemaUi,
} from './CreateForm/AANZFTACOOSchema';

export const Issue = () => {
  // TODO: Request forms from the backend.
  const forms = [
    {
      id: '001',
      name: 'AANZFTA-COO',
      fullForm: { schema: fullCooSchema, uiSchema: fullCooSchemaUi },
      partialForm: { schema: partialCooSchema, uiSchema: partialCooSchemaUi },
    },
  ];

  const navigate = useNavigate();

  const goToForm = (form: Form, credentialType: any) => {
    navigate('/form', {
      state: {
        form,
        credentialType,
      },
    });
  };

  const handleFormSelected = (selectedForm: FormOption) => {
    const { partialForm, fullForm, formType, credentialType } = selectedForm;

    goToForm(
      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
      formType === 'partial' ? (partialForm as Form) : (fullForm as Form),
      credentialType
    );
  };

  return (
    <BaseLayout title="Form Select">
      <Text variant="h4" fontWeight="bold" paddingBottom="24px">
        Issue document
      </Text>
      <Text>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis
        molestie, dictum est a, mattis tellus. Sed dignissim, metus nec
        fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus
        elit sed risus. Maecenas eget condimentum velit, sit amet feugiat
        lectus. Class aptent taciti sociosqu ad litora torquent per conubia
        nostra, per inceptos himenaeos. Praesent auctor purus luctus enim
        egestas, ac scelerisque ante pulvinar. Donec ut rhoncus ex. Suspendisse
        ac rhoncus nisl, eu tempor urna. Curabitur vel bibendum lorem. Morbi
        convallis convallis diam sit amet lacinia. Aliquam in elementum tellus.
      </Text>
      <Grid container>
        <Grid item display={'flex'} justifyContent={'center'} width={'100%'}>
          <Paper
            sx={{
              width: '100%',
              maxWidth: '38rem',
              padding: { xs: '20px', md: '56px' },
            }}
          >
            <VCOptions forms={forms} onFormSelected={handleFormSelected} />
          </Paper>
        </Grid>
      </Grid>
    </BaseLayout>
  );
};
