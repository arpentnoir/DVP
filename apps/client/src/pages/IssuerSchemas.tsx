import { Card, Text } from '@dvp/vc-ui';
import { Box, Stack, Switch } from '@mui/material';
import { useParams } from 'react-router-dom';
import { BaseLayout } from '../layouts';

// TODO: Replace with real data
const schemas = [
  {
    name: 'Schema 1',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
  },
  {
    name: 'Schema 2',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
  },
  {
    name: 'Schema 3',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
  },
];

export const IssuerSchemas = () => {
  const { issuerName } = useParams();

  return (
    <BaseLayout title="Issuer Schemas">
      <Box paddingBottom="40px">
        <Text variant="h4" fontWeight="bold">
          {issuerName}
        </Text>
      </Box>

      <Box>
        <Text variant="body1" fontWeight="bold">
          Document schemas
        </Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</Text>
        <Stack
          direction={{ xs: 'column', sm: 'row' }}
          spacing="8px"
          paddingTop="40px"
        >
          {schemas.map((schema) => (
            <Card
              key={schema.name}
              name={schema.name}
              text={schema.text}
              headerAction={<Switch />}
            />
          ))}
        </Stack>
      </Box>
    </BaseLayout>
  );
};
