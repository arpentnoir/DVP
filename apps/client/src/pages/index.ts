export * from './CreateForm';
export * from './Home';
export * from './Verify';
export * from './Viewer';
export * from './Issue';
