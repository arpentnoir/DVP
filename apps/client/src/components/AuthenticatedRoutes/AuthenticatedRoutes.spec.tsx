import { render } from '@testing-library/react';
import { useEffect } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Routes,
  useNavigate,
} from 'react-router-dom';
import * as pages from '../../pages';
import { Home } from '../../pages';
import { AuthenticatedRoutes } from './AuthenticatedRoutes';

describe('BaseApp', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<AuthenticatedRoutes />);

    expect(baseElement).toMatchSnapshot();
  });
  it('should not show authenticated routes', () => {
    (jest.spyOn(pages, 'Home') as jest.SpyInstance).mockImplementation(() => {
      const navigate = useNavigate();
      useEffect(() => {
        navigate('/authed');
      }, []);
    });

    const { queryByText } = render(
      <Router>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route element={<AuthenticatedRoutes />}>
            <Route path="/authed" element={<div>authed route</div>} />
          </Route>
        </Routes>
      </Router>
    );
    const authedElement = queryByText('authed route');
    expect(authedElement).toBeNull();
  });
});
