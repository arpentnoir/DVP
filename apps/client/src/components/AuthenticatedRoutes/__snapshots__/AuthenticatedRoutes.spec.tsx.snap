// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseApp should render correctly 1`] = `
<body>
  <div>
    <div
      style="height: 100%;"
    />
  </div>
</body>
`;
